spec/controllers/tasks_controller_spec.rb

```diff
@@ -233,6 +233,17 @@
         post_request
         expect(response).to render_template('new')
       end
+
+      context 'when new label needs to be created' do
+        let(:invalid_attributes) { {title: '', label_names: [not_existing_label_name]} }
+        let(:not_existing_label_name) { 'some new label' }
+
+        it "re-renders the 'new' template successfully" do
```

Comment:
In this test, we should also check that the `not_existing_label_name` is present and *checked* (Hint: Write a test that fails without the `@task.labels.map(&:name))` change, but only succeeds otherwise. If you test this manually, you'll quickly notice the difference in the HTML -- that is the `selected` attribute for this `option`).

Reply:
I think that would be a good integration test. Something like this:
```
expect(rendered).to have_selector('option[value="some new label"][selected="selected"]')
```

Other than this I found rails-dom-testing to be in the Gemfile but could not find any usages in CodeHarbor and also no examples of how to use it online.
The last option I could think of is using regex again although that is probably not a good idea.

Reply:
Indeed, I would also go for an integration test. Let's use this as an example and follow up on it next week ;).

The `rails-dom-testing` is only included as an indirect dependency in the `Gemfile.lock`, and therefore not (directly) used so far.